src/stream/src/executor/top_n_local.rs

```diff
@@ -0,0 +1,479 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use async_trait::async_trait;
+use risingwave_common::array::{Op, Row, StreamChunk};
+use risingwave_common::catalog::{Schema, TableId};
+use risingwave_common::types::DataType;
+use risingwave_common::util::ordered::{OrderedRow, OrderedRowDeserializer};
+use risingwave_common::util::sort_util::{OrderPair, OrderType};
+use risingwave_storage::StateStore;
+
+use super::error::StreamExecutorResult;
+use super::managed_state::top_n::{ManagedTopNStateNew, TopNStateRow};
+use super::top_n_executor::{generate_output, TopNExecutorBase, TopNExecutorWrapper};
+use super::{BoxedMessageStream, Executor, ExecutorInfo, PkIndices, PkIndicesRef};
+
+pub type LocalTopNExecutor<S> = TopNExecutorWrapper<InnerLocalTopNExecutorNew<S>>;
+
+impl<S: StateStore> LocalTopNExecutor<S> {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        input: Box<dyn Executor>,
+        order_pairs: Vec<OrderPair>,
+        offset_and_limit: (usize, Option<usize>),
+        pk_indices: PkIndices,
+        store: S,
+        table_id: TableId,
+        total_count: usize,
+        executor_id: u64,
+        key_indices: Vec<usize>,
+        group_by: usize,
```

Comment:
Add

```
// TODO: support multiple group keys
```

Reply:
is there any problem supporting multiple group keys in this PR? I think maybe it is ok?

Reply:
let me try